code/game/machinery/washing_machine.dm

```diff
@@ -57,283 +118,307 @@
+					to_chat(user, "<span class='warning'>You need a better grip to do that!</span>")
+				else
+					visible_message("<span class='notice'>[user] starts putting [G.affecting.name] into [src].</span>")
+					if(do_after(user, 20, target = G.affecting))
+						visible_message("<span class='warning'>[user] puts [G.affecting.name] into [src].</span>")
+						add_fingerprint(user)
+						G.affecting.forceMove(src)
+						qdel(G)
+						full = TRUE
 			update_icon()
 		else
+			if(!open)
+				to_chat(user, "<span class='notice'>[src]'s door is closed.</span>")
+			else
+				if(full)
+					to_chat(user, "<span class='notice'>[src] is already full.</span>")
+				else
+					to_chat(user, "<span class='notice'>You can't fit [W] in [src].</span>")
 			return ..()
-	else if(istype(W,/obj/item/stack/sheet/hairlesshide) || \
-		istype(W,/obj/item/clothing/under) || \
-		istype(W,/obj/item/clothing/mask) || \
-		istype(W,/obj/item/clothing/head) || \
-		istype(W,/obj/item/clothing/gloves) || \
-		istype(W,/obj/item/clothing/shoes) || \
-		istype(W,/obj/item/clothing/suit) || \
-		istype(W,/obj/item/bedsheet))
-
-		//YES, it's hardcoded... saves a var/can_be_washed for every single clothing item.
-		if( istype(W,/obj/item/clothing/under/plasmaman ) )
-			to_chat(user, "This item does not fit.")
-			return
-		if( istype(W,/obj/item/clothing/suit/space ) )
-			to_chat(user, "This item does not fit.")
-			return
-		if( istype(W,/obj/item/clothing/suit/syndicatefake ) )
-			to_chat(user, "This item does not fit.")
-			return
-//		if( istype(W,/obj/item/clothing/suit/powered ) )
-//			to_chat(user, "This item does not fit.")
-//			return
-		if( istype(W,/obj/item/clothing/suit/cyborg_suit ) )
-			to_chat(user, "This item does not fit.")
-			return
-		if( istype(W,/obj/item/clothing/suit/bomb_suit ) )
-			to_chat(user, "This item does not fit.")
-			return
-		if( istype(W,/obj/item/clothing/suit/armor ) )
-			to_chat(user, "This item does not fit.")
-			return
-		if( istype(W,/obj/item/clothing/mask/gas ) )
-			to_chat(user, "This item does not fit.")
-			return
-		if( istype(W,/obj/item/clothing/mask/cigarette ) )
-			to_chat(user, "This item does not fit.")
-			return
-		if( istype(W,/obj/item/clothing/head/syndicatefake ) )
-			to_chat(user, "This item does not fit.")
-			return
-//		if( istype(W,/obj/item/clothing/head/powered ) )
-//			to_chat(user, "This item does not fit.")
-//			return
-		if( istype(W,/obj/item/clothing/head/helmet ) )
-			to_chat(user, "This item does not fit.")
-			return
-		if( istype(W,/obj/item/clothing/gloves/furgloves ) )
-			to_chat(user, "This item does not fit.")
-			return
+	else
+		var/list/allowed = list(/obj/item/stack/sheet/hairlesshide, W,/obj/item/clothing/under, /obj/item/clothing/mask, /obj/item/clothing/head, /obj/item/clothing/gloves, /obj/item/clothing/shoes, /obj/item/clothing/suit, /obj/item/bedsheet, /obj/item/storage/wallet)
```

Comment:
this should all be moved out of the attackby proc. Keep your proc functions doing mostly only what they're intended for, attackby() is for deciding what to do when someone attack the washer with hands, grabs, items etc, it's not for performing the function of the object it's overriding.